extensions/kafka/src/main/java/io/deephaven/kafka/SimpleImpl.java

```diff
@@ -3,42 +3,398 @@
+import org.apache.kafka.common.serialization.BytesSerializer;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.DoubleDeserializer;
+import org.apache.kafka.common.serialization.DoubleSerializer;
+import org.apache.kafka.common.serialization.FloatDeserializer;
+import org.apache.kafka.common.serialization.FloatSerializer;
+import org.apache.kafka.common.serialization.IntegerDeserializer;
+import org.apache.kafka.common.serialization.IntegerSerializer;
+import org.apache.kafka.common.serialization.LongDeserializer;
+import org.apache.kafka.common.serialization.LongSerializer;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.serialization.ShortDeserializer;
+import org.apache.kafka.common.serialization.ShortSerializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.kafka.common.serialization.UUIDDeserializer;
+import org.apache.kafka.common.serialization.UUIDSerializer;
+import org.apache.kafka.common.utils.Bytes;
+import org.jetbrains.annotations.NotNull;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+
+import static io.deephaven.kafka.KafkaTools.KEY_COLUMN_NAME_DEFAULT;
+import static io.deephaven.kafka.KafkaTools.KEY_COLUMN_NAME_PROPERTY;
+import static io.deephaven.kafka.KafkaTools.KEY_COLUMN_TYPE_PROPERTY;
+import static io.deephaven.kafka.KafkaTools.VALUE_COLUMN_NAME_DEFAULT;
+import static io.deephaven.kafka.KafkaTools.VALUE_COLUMN_NAME_PROPERTY;
+import static io.deephaven.kafka.KafkaTools.VALUE_COLUMN_TYPE_PROPERTY;
 
 class SimpleImpl {
+
     /**
      * Single spec for unidimensional (basic Kafka encoded for one type) fields.
      */
     static final class SimpleConsume extends Consume.KeyOrValueSpec {
-        final String columnName;
-        final Class<?> dataType;
+        private final String columnName;
+        private final Class<?> dataType;
 
         SimpleConsume(final String columnName, final Class<?> dataType) {
             this.columnName = columnName;
             this.dataType = dataType;
         }
 
         @Override
-        DataFormat dataFormat() {
-            return DataFormat.SIMPLE;
+        boolean isIgnore() {
+            return false;
+        }
+
+        @Override
+        public Optional<SchemaProvider> schemaProvider() {
+            return Optional.empty();
+        }
+
+        @Override
+        Deserializer<?> deserializer(
+                KeyOrValue keyOrValue,
+                SchemaRegistryClient schemaRegistryClient,
+                Map<String, ?> configs) {
+            final Type<?> type = getType(keyOrValue, configs);
+            final Deserializer<?> deserializer = SerDeserVisitor.deserializer(type).orElse(null);
+            if (deserializer != null) {
```

Comment:
If the user specifies deserializer in properties... we should probably be using it, and verifying it matches the type.